### PR TITLE
Solve Issue #73: automatic resample extraction

### PR DESCRIPTION
**Pull request by jakob-r**

### CODE REVIEW

File: R/getResampleExtract.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
+getResampleExtract2 = function(learner) {
+  UseMethod("getResampleExtract2")
+}
+
+getResampleExtract2.NULL = function(learner) {
```
Comment (mllg) [inline R/getResampleExtract.R]: This should be `getResampleExtract2.default`.

File: R/getResampleExtract.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+getResampleExtract2.FeatSelWrapper = function(learner) {
+  c(list(FeatSelResult = getFeatSelResult), getResampleExtract2(learner$next.learner))
+}
+
+getResampleExtract2.FilterWrapper = function(learner) {
+  c(list(FilteredFeatures = getFilteredFeatures), getResampleExtract2(learner$next.learner))
+}
+
+getResampleExtract = function(learner){
```
Comment (mllg) [inline R/getResampleExtract.R]: What are you trying to do here? This seems pretty odd.

File: tests/testthat/test_base_TuneWrapper.R [hunk 3]
```diff
@@ -36,8 +36,9 @@ test_that("TuneWrapper", {
   lrn1 = makeLearner("classif.ksvm", predict.type = "prob")
   lrn2 = makeTuneWrapper(lrn1, resampling = makeResampleDesc("Holdout"), par.set = ps1, control = makeTuneControlGrid())
   expect_equal(lrn2$predict.type, "prob")
-  r = resample(lrn2, binaryclass.task, makeResampleDesc("Holdout"), measures = mlr::auc)
+  r = resample(lrn2, binaryclass.task, makeResampleDesc("CV"), measures = mlr::auc)
```
Comment (mllg) [inline tests/testthat/test_base_TuneWrapper.R]: Try to keep the unit tests as inexpensive as possible. Better use Holdout or CV with iters = 2.

Comment (jakob-r) [inline tests/testthat/test_base_TuneWrapper.R]: My mistake - I changed it just to see what happens. forgot to change it back

File: R/getResampleExtract.R (hunk 1 above)
Comment (jakob-r) [inline R/getResampleExtract.R]: will change

File: R/getResampleExtract.R (hunk 2 above)
Comment (jakob-r) [inline R/getResampleExtract.R]: We find out what we can extract by default for each wrapper. For multiple wrapped learners we can extract more than just one information so we collect all usable functions and then build one function which returns a list. Compare with [this tutorial part](http://mlr-org.github.io/mlr-tutorial/devel/html/nested_resampling/index.html#filter-methods-with-tuning).

Comment (mllg) [inline R/getResampleExtract.R]: Well okay, then throw in some comments please. The `force` is not required btw.

**Comment by jakob-r** [comment]
I am not sure what's the best way to test this. 
a) Test if the extract works in any function wich have `extract` as an argument
b) Collect all possible functions in one file and test if extracts works as expected
c) Do not write additional test to save execution time

**Comment by jakob-r** [comment]
d) put the extract tests only there were we use the things (Wrappers) we want to extract things from

**Comment by jakob-r** [comment]
Additionally:
When we use multiple Wrappers, by default, a list with all different extractions is returned. However if we only have one Wrapper which offers an extraction at the moment we don't put the result in an extra list (with only one item)
Somehow I don't like this inconsistency: Not defined if result is list or single item. What is the consensus here?

File: R/getResampleExtract.R (hunk 2 above)
Comment (jakob-r) [inline R/getResampleExtract.R]: added comments on top of file

**Pull request merged**